### PR TITLE
bpo-44940: Clarify the documentation of re.findall()

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue44940

### CODE REVIEW

**Comment by serhiy-storchaka** [comment]
@rhettinger, please take a look. The current documentation is not clear for beginners, and I know that you are interesting in teaching. Here is an attempt to make it more explicit, but I am not sure about my English.

**Comment by vedgar** [comment]
Very nice. I'd say "that group" instead of "the group" in "only 1 group" case, and at the end, "doesn't affect the _form of_ the result" (I mean, obviously they can affect the result itself.:)

File: Doc/library/re.rst [hunk 1]
```diff
@@ -824,10 +824,20 @@ form.
...
-   the order found.  If one or more groups are present in the pattern, return a
-   list of groups; this will be a list of tuples if the pattern has more than
-   one group.  Empty matches are included in the result.
+   strings or tuples.  The *string* is scanned left-to-right, and matches
+   are returned in the order found.  Empty matches are included in the result.
+
+   The result depends on the number of capturing groups in the pattern.
+   If there are no groups, return a list of strings matching the whole
+   pattern.  If there is exactly one group, return a list of strings
+   matching the group.  If more than one groups are present, return a list
```
Comment (Fidget-Spinner) [inline Doc/library/re.rst]: Ditto to what vedgar said above about `that`. These are just minor suggestions, feel free to ignore.
```suggestion
   matching that group.  If multiple groups are present, return a list
```

File: Doc/library/re.rst [hunk 2]
```diff
@@ -824,10 +824,20 @@ form.
...
+   of tuples of strings matching the groups.  Non-capturing groups do not
+   affect the result.
+
+      >>> re.findall(r'\bf[a-z]*', 'which foot or hand fell fastest')
+      ['foot', 'fell', 'fastest']
+      >>> re.findall(r'(\w+)=(\d+)', 'set width=20 and height=10')
```
Comment (Fidget-Spinner) [inline Doc/library/re.rst:839]: Great example! The sentence above was a little hard to parse. But this example clears everything up.

**Review by Fidget-Spinner — APPROVED** [review]
I'm not Raymond, but from a `re` user's perspective, this is much better than the old one. Thanks Serhiy. The English LGTM too.

File: Doc/library/re.rst [hunk 3]
```diff
@@ -824,10 +824,20 @@ form.
...
-   one group.  Empty matches are included in the result.
+   strings or tuples.  The *string* is scanned left-to-right, and matches
+   are returned in the order found.  Empty matches are included in the result.
+
+   The result depends on the number of capturing groups in the pattern.
+   If there are no groups, return a list of strings matching the whole
+   pattern.  If there is exactly one group, return a list of strings
+   matching the group.  If more than one groups are present, return a list
+   of tuples of strings matching the groups.  Non-capturing groups do not
+   affect the result.
```
Comment (serhiy-storchaka) [inline Doc/library/re.rst]:
```suggestion
   affect the form of the result.
```

**Comment by serhiy-storchaka** [comment]
Thank you, @vedgar and @Fidget-Spinner.
